### PR TITLE
Add description for `tidb_auth_token` authentication

### PR DESCRIPTION
**Pull request by CbcWestwolf**
### What is changed, added or deleted? (Required)

Add description for `tidb_auth_token`, which can be used not only inside TiDB Cloud, but also by normal tidb users.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/issues/38504

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by CbcWestwolf** [comment]
/cc bb7133 
PTAL

File: security-compatibility-with-mysql.md [hunk 1]
```diff
@@ -108,9 +108,9 @@ TiDB 的密码重用策略功能与 MySQL 一致，在实现密码重用策略
 
 ## 可用的身份验证插件
 
-TiDB 支持多种身份验证方式。通过使用 [`CREATE USER`](/sql-statements/sql-statement-create-user.md) 语句和 [`ALTER USER`](/sql-statements/sql-statement-create-user.md) 语句，即可创建新用户或更改 TiDB 权限系统内的已有用户。TiDB 身份验证方式与 MySQL 兼容，其名称与 MySQL 保持一致。
+TiDB 支持多种身份验证方式。通过使用 [`CREATE USER`](/sql-statements/sql-statement-create-user.md) 语句和 [`ALTER USER`](/sql-statements/sql-statement-alter-user.md) 语句，即可创建新用户或更改 TiDB 权限系统内的已有用户。TiDB 身份验证方式与 MySQL 兼容，其名称与 MySQL 保持一致。
 
-TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务器和客户端建立连接时，如要指定服务器对外通告的默认验证方式，可通过 [`default_authentication_plugin`](/system-variables.md#default_authentication_plugin) 变量进行设置。`tidb_sm3_password` 为仅在 TiDB 支持的 SM3 身份验证方式，使用该方式登录的用户需要使用 [TiDB-JDBC](https://github.com/pingcap/mysql-connector-j/tree/release/8.0-sm3)。`tidb_auth_token` 为仅用于 TiDB Cloud 内部的基于 JSON Web Token (JWT) 的认证方式。
+TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务器和客户端建立连接时，如要指定服务器对外通告的默认验证方式，可通过 [`default_authentication_plugin`](/system-variables.md#default_authentication_plugin) 变量进行设置。`tidb_sm3_password` 为仅在 TiDB 支持的 SM3 身份验证方式，使用该方式登录的用户需要使用 [TiDB-JDBC](https://github.com/pingcap/mysql-connector-j/tree/release/8.0-sm3)。
```
Comment (bb7133) [inline security-compatibility-with-mysql.md:113]: 这里建议还是保留 tidb_auth_token 的描述吧，可以说 JWT 认证在 TiDB Cloud 上使用，同时也可以自行在 OP 环境使用？

File: security-compatibility-with-mysql.md [hunk 2]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
 | ed25519 (MariaDB)        | 否               |
 | GSSAPI (MariaDB)         | 否               |
 | FIDO                     | 否               |
+
+### `tidb_auth_token`
+
+`tidb_auth_token` 是一种基于 [JSON Web Token (JWT)](https://datatracker.ietf.org/doc/html/rfc7519) 的无密码认证方式，用于 TiDB Cloud 内部用户认证。不同于 `mysql_native_passsword`、`caching_sha2_password` 等使用密码的认证方式，创建 `tidb_auth_token` 用户时无需设置并保存自定义的密码；使用 `tidb_auth_token` 进行登录时使用签发的 token 进行登录，可以简化用户的认证过程并提升安全性。
+
+JWT 由 header、payload 和 signature 三部分组成，每部分使用 base64 编码之后拼接成一个字符串，中间用点号（`.`）分开。
```
Comment (bb7133) [inline security-compatibility-with-mysql.md:138]: 文档中混用了 'header'、'payload' 这些英文的大小写（首行大写），我感觉放在中文文档里面这么用比较怪异，是不是全首字母大写比较好？可以请文档的同事提一下建议。

File: security-compatibility-with-mysql.md [hunk 3]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
 | ed25519 (MariaDB)        | 否               |
 | GSSAPI (MariaDB)         | 否               |
 | FIDO                     | 否               |
+
+### `tidb_auth_token`
+
+`tidb_auth_token` 是一种基于 [JSON Web Token (JWT)](https://datatracker.ietf.org/doc/html/rfc7519) 的无密码认证方式，用于 TiDB Cloud 内部用户认证。不同于 `mysql_native_passsword`、`caching_sha2_password` 等使用密码的认证方式，创建 `tidb_auth_token` 用户时无需设置并保存自定义的密码；使用 `tidb_auth_token` 进行登录时使用签发的 token 进行登录，可以简化用户的认证过程并提升安全性。
```
Comment (bb7133) [inline security-compatibility-with-mysql.md:136]: > 用于 TiDB Cloud 内部用户认证

comment 同上

File: security-compatibility-with-mysql.md [hunk 4]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+复制上面最后一行的 token 用于登录：
+
+```Shell
+mycli -h 127.0.0.1 -P 4000 -u 'user@pingcap.com' -p '<the-token-generated>'
+```
+
+注意这里使用的 mysql 客户端必须支持 `mysql_clear_password` 插件。[mycli](https://www.mycli.net/) 默认开启这一插件，如果使用 [mysql 命令行客户端](https://dev.mysql.com/doc/refman/8.0/en/mysql.html) 则需要 `--enable-cleartext-plugin` 选项来开启这个插件：
+
+```Shell
+mysql -h 127.0.0.1 -P 4000 -u 'user@pingcap.com' -p '<the-token-generated>' --enable-cleartext-plugin
```
Comment (bb7133) [inline security-compatibility-with-mysql.md:254]:
```suggestion
mysql -h 127.0.0.1 -P 4000 -u 'user@pingcap.com' -p'<the-token-generated>' --enable-cleartext-plugin
```
https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_password

> If given, there must be no space between [--password=](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_password) or -p and the password following it. If no password option is specified, the default is to send no password.

Please remove the space here.

File: security-compatibility-with-mysql.md [hunk 5]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+Payload 是 JWT 的主体部分，保存用户的信息，每个字段就是一个 claim（声明）。TiDB 用户认证要求的几个声明如下：
+
+* `iss`：如果[创建用户](/sql-statements/sql-statement-create-user.md)时未指定 `TOKEN_ISSUER` 或者设置为空串，则可以不包含该声明；否则应该与设置值相同
+* `sub`：TiDB 中要求该值与待认证的用户名相同
+* `iat`：发布时间戳。TiDB 中要求该值不得晚于认证时的时间，不得早于认证前 15 分钟
+* `exp`：到期时间戳。如果认证时已经到期，则认证失败
+
+此外 TiDB 中还要求包含
+
+* `email`：邮件地址。创建用户时可以通过 `ATTRIBUTE '{"email": "xxxx@pingcap.com"}` 指定 email 信息。如果创建用户时未指定 email 信息，则该声明应设置为空串；否则应该与设置值相同
```
Comment (bb7133) [inline security-compatibility-with-mysql.md:165]:
```suggestion
* `email`：邮件地址。创建用户时可以通过 `ATTRIBUTE '{"email": "xxxx@pingcap.com"}'` 指定 email 信息。如果创建用户时未指定 email 信息，则该声明应设置为空串；否则应该与设置值相同
```

File: security-compatibility-with-mysql.md [hunk 6]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+
+`tidb_auth_token` 是一种基于 [JSON Web Token (JWT)](https://datatracker.ietf.org/doc/html/rfc7519) 的无密码认证方式，用于 TiDB Cloud 内部的用户认证，用户通过配置也可以在自托管环境使用。不同于 `mysql_native_passsword`、`caching_sha2_password` 等使用密码的认证方式，创建 `tidb_auth_token` 用户时无需设置并保存自定义的密码；使用 `tidb_auth_token` 进行登录时使用签发的 token 进行登录，可以简化用户的认证过程并提升安全性。
+
+JWT 由 Header、Payload 和 Signature 三部分组成，每部分使用 base64 编码之后拼接成一个字符串，中间用点号（`.`）分开。
+
+Header 描述 JWT 的元数据，包含 3 个属性：
+
+* `alg`：表示签名使用的算法，默认为 `RS256`
+* `typ`：表示 token 的类型，统一为 `JWT`
+* `kid`：表示用于生成 token 签名的 key id
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:144]:
```suggestion
* `alg`：表示签名使用的算法，默认为 `RS256`。
* `typ`：表示 token 的类型，为 `JWT`。
* `kid`：表示用于生成 token 签名的 key ID。
```

File: security-compatibility-with-mysql.md [hunk 7]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+
+```json
+{
+  "alg": "RS256",
+  "kid": "the-key-id-0",
+  "typ": "JWT"
+}
+```
+
+Payload 是 JWT 的主体部分，保存用户的信息，每个字段就是一个 claim（声明）。TiDB 用户认证要求的几个声明如下：
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:156]:
```suggestion
Payload 是 JWT 的主体部分，用于保存用户的信息。Payload 中的每个字段称为一个 claim（声明）。TiDB 用户认证要求提供的声明如下：
```

File: security-compatibility-with-mysql.md [hunk 8]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+
+* `iss`：如果[创建用户](/sql-statements/sql-statement-create-user.md)时未指定 `TOKEN_ISSUER` 或者设置为空串，则可以不包含该声明；否则应该与设置值相同
+* `sub`：TiDB 中要求该值与待认证的用户名相同
+* `iat`：发布时间戳。TiDB 中要求该值不得晚于认证时的时间，不得早于认证前 15 分钟
+* `exp`：到期时间戳。如果认证时已经到期，则认证失败
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:161]:
```suggestion
* `iss`：如果[创建用户](/sql-statements/sql-statement-create-user.md)时未指定 `TOKEN_ISSUER` 或者将其设置为了空字符串，则可以不包含该声明；否则 `iss` 应该与 `TOKEN_ISSUER` 设置值相同。
* `sub`：TiDB 中要求该值与待认证的用户名相同。
* `iat`：发布 token 的时间戳。TiDB 中要求该值不得晚于认证时的时间，也不得早于认证前 15 分钟。
* `exp`：token 到期的时间戳。如果 token 在认证时已经过期，则认证失败。
```

File: security-compatibility-with-mysql.md [hunk 9]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+
+此外 TiDB 中还要求包含
+
+* `email`：邮件地址。创建用户时可以通过 `ATTRIBUTE '{"email": "xxxx@pingcap.com"}'` 指定 email 信息。如果创建用户时未指定 email 信息，则该声明应设置为空串；否则应该与设置值相同
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:165]:
```suggestion
* `email`：邮件地址。创建用户时可以通过 `ATTRIBUTE '{"email": "xxxx@pingcap.com"}'` 指定 email 信息。如果创建用户时未指定 email 信息，则该声明应设置为空字符串；否则该声明应该与创建用户时的设置值相同。
```

File: security-compatibility-with-mysql.md [hunk 10]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+
+下面是几个合法的 Payload 示例：
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:167]:
```suggestion
Payload 示例：
```

File: security-compatibility-with-mysql.md [hunk 11]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+
+```json
+{
+  "email": "",
+  "exp": 1703305494,
+  "iat": 1703304594,
+  "sub": "user@pingcap.com"
+}
+```
+
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:189]:
```suggestion
```

Comment (qiancai) [reply security-compatibility-with-mysql.md:189]: 这个例子也可以不要？

File: security-compatibility-with-mysql.md [hunk 12]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+Signature 对上面两部分数据进行签名。
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:190]:
```suggestion
Signature 用于对 Header 和 Payload 这两部分数据进行签名。
```

File: security-compatibility-with-mysql.md [hunk 13]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+
+> **警告：**
+>
+> 1. Header 与 Payload 使用 base64 进行编码的过程是可逆的，请勿在 Payload 中携带敏感数据
+> 2. `tidb_auth_token` 认证方式要求客户端支持 [`mysql_clear_password`](https://dev.mysql.com/doc/refman/8.0/en/cleartext-pluggable-authentication.html) 插件，并将 token 以明文的方式发送至 TiDB，因此请[为 TiDB 开启加密传输](/enable-tls-between-clients-and-servers.md) 后再使用 `tidb_auth_token` 进行认证
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:195]:
```suggestion
> - Header 与 Payload 使用 base64 进行编码的过程是可逆的，请勿在 Payload 中携带敏感数据。
> - `tidb_auth_token` 认证方式要求客户端支持 [`mysql_clear_password`](https://dev.mysql.com/doc/refman/8.0/en/cleartext-pluggable-authentication.html) 插件，并将 token 以明文的方式发送至 TiDB，因此请[为 TiDB 开启加密传输](/enable-tls-between-clients-and-servers.md) 后再使用 `tidb_auth_token` 进行认证。
```

File: security-compatibility-with-mysql.md [hunk 14]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+
+配置并使用 `tidb_auth_token` 作为 TiDB 用户的认证方式，有以下几个步骤：
+
+1. 在 TiDB 配置文件中设置 [`auth-token-jwks`](/tidb-configuration-file.md#auth-token-jwks-从-v640-版本开始引入) 和 [`auth-token-refresh-interval`](/tidb-configuration-file.md#auth-token-refresh-interval-从-v640-版本开始引入)
+2. 定期更新保存 JWKS 至 `auth-token-jwks` 指定的路径
+3. 创建使用 `tidb_auth_token` 认证的用户，并根据需要通过 `REQUIRE TOKEN_ISSUER` 和 `ATTRIBUTE '{"email": "xxxx@pingcap.com"}` 指定 `iss` 与 `email` 信息
+4. 生成并签发用于认证的 token，通过 mysql 客户端的 `mysql_clear_text` 插件进行认证
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:202]:
```suggestion
1. 在 TiDB 配置文件中设置 [`auth-token-jwks`](/tidb-configuration-file.md#auth-token-jwks-从-v640-版本开始引入) 和 [`auth-token-refresh-interval`](/tidb-configuration-file.md#auth-token-refresh-interval-从-v640-版本开始引入)。
2. 定期更新保存 JWKS 至 `auth-token-jwks` 指定的路径。
3. 创建使用 `tidb_auth_token` 认证的用户，并根据需要通过 `REQUIRE TOKEN_ISSUER` 和 `ATTRIBUTE '{"email": "xxxx@pingcap.com"}` 指定 `iss` 与 `email` 信息。
4. 生成并签发用于认证的 token，通过 mysql 客户端的 `mysql_clear_text` 插件进行认证。
```

File: security-compatibility-with-mysql.md [hunk 15]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+
+1. 在 TiDB 配置文件中设置 [`auth-token-jwks`](/tidb-configuration-file.md#auth-token-jwks-从-v640-版本开始引入) 和 [`auth-token-refresh-interval`](/tidb-configuration-file.md#auth-token-refresh-interval-从-v640-版本开始引入)
+2. 定期更新保存 JWKS 至 `auth-token-jwks` 指定的路径
+3. 创建使用 `tidb_auth_token` 认证的用户，并根据需要通过 `REQUIRE TOKEN_ISSUER` 和 `ATTRIBUTE '{"email": "xxxx@pingcap.com"}` 指定 `iss` 与 `email` 信息
+4. 生成并签发用于认证的 token，通过 mysql 客户端的 `mysql_clear_text` 插件进行认证 
+
+#### 示例
+
+1. 通过 `go install github.com/cbcwestwolf/generate_jwt` 安装 JWT 生成工具。该工具仅用于生成测试 `tidb_auth_token` 的 JWT。
+2. 获取示例 JWKS：`wget https://raw.githubusercontent.com/CbcWestwolf/generate_jwt/master/JWKS.json`
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:207]:
```suggestion
2. 获取示例 JWKS：

    ```bash
    wget https://raw.githubusercontent.com/CbcWestwolf/generate_jwt/master/JWKS.json
    ```

```

File: security-compatibility-with-mysql.md [hunk 16]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+3. 在 TiDB 的配置文件 `config.toml` 中配置上述 JWKS 文件的路径
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:208]:
```suggestion
3. 在 TiDB 的配置文件 `config.toml` 中配置上述 JWKS 文件的路径：
```

File: security-compatibility-with-mysql.md [hunk 17]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+
+    ```toml
+    [security]
+    auth-token-jwks = "JWKS.json"
+    ```
+
+4. 启动 `tidb-server`
+5. 创建使用 `tidb_auth_token` 认证的用户 `user@pingcap.com`
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:216]:
```suggestion
4. 启动 `tidb-server`。
5. 创建使用 `tidb_auth_token` 认证的用户 `user@pingcap.com`：
```

File: security-compatibility-with-mysql.md [hunk 18]
```diff
@@ -130,3 +130,130 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
 | ed25519 (MariaDB)        | 否               |
 | GSSAPI (MariaDB)         | 否               |
 | FIDO                     | 否               |
+
+### `tidb_auth_token`
+
+`tidb_auth_token` 是一种基于 [JSON Web Token (JWT)](https://datatracker.ietf.org/doc/html/rfc7519) 的无密码认证方式，用于 TiDB Cloud 内部的用户认证，用户通过配置也可以在自托管环境使用。不同于 `mysql_native_passsword`、`caching_sha2_password` 等使用密码的认证方式，创建 `tidb_auth_token` 用户时无需设置并保存自定义的密码；使用 `tidb_auth_token` 进行登录时使用签发的 token 进行登录，可以简化用户的认证过程并提升安全性。
+
+JWT 由 Header、Payload 和 Signature 三部分组成，每部分使用 base64 编码之后拼接成一个字符串，中间用点号（`.`）分开。
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:138]:
```suggestion
JWT 由 Header、Payload 和 Signature 三部分组成。这三部分通过 base64 编码后，使用点号（`.`）拼接成一个字符串，以便在客户端和服务器之间传输。
```

File: security-compatibility-with-mysql.md [hunk 19]
```diff
@@ -130,3 +130,121 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
 | ed25519 (MariaDB)        | 否               |
 | GSSAPI (MariaDB)         | 否               |
 | FIDO                     | 否               |
+
+### `tidb_auth_token`
+
+`tidb_auth_token` 是一种基于 [JSON Web Token (JWT)](https://datatracker.ietf.org/doc/html/rfc7519) 的无密码认证方式，用于 TiDB Cloud 内部的用户认证，用户通过配置也可以在自托管环境使用。不同于 `mysql_native_passsword`、`caching_sha2_password` 等使用密码的认证方式，`tidb_auth_token` 认证方式在创建用户时无需设置并保存自定义密码，在用户登录时只需使用一个签发的 token，从而简化用户的认证过程并提升安全性。
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:136]:
```suggestion
`tidb_auth_token` 是一种基于 [JSON Web Token (JWT)](https://datatracker.ietf.org/doc/html/rfc7519) 的无密码认证方式。在 v6.4.0 中， `tidb_auth_token` 仅用于 TiDB Cloud 内部的用户认证，从 v6.5.0 起，你也可以将 `tidb_auth_token` 配置为 TiDB 自托管环境中用户的认证方式。不同于 `mysql_native_passsword`、`caching_sha2_password` 等使用密码的认证方式，`tidb_auth_token` 认证方式在创建用户时无需设置并保存自定义密码，在用户登录时只需使用一个签发的 token，从而简化用户的认证过程并提升安全性。
```

File: security-compatibility-with-mysql.md [hunk 20]
```diff
@@ -130,3 +130,121 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+> - Header 与 Payload 使用 base64 进行编码的过程是可逆的，请勿在 Payload 中携带敏感数据。
+> - `tidb_auth_token` 认证方式要求客户端支持 [`mysql_clear_password`](https://dev.mysql.com/doc/refman/8.0/en/cleartext-pluggable-authentication.html) 插件，并将 token 以明文的方式发送至 TiDB，因此请[为 TiDB 开启加密传输](/enable-tls-between-clients-and-servers.md) 后再使用 `tidb_auth_token` 进行认证。
+
+配置并使用 `tidb_auth_token` 作为 TiDB 用户的认证方式，有以下几个步骤：
+
+1. 在 TiDB 配置文件中设置 [`auth-token-jwks`](/tidb-configuration-file.md#auth-token-jwks-从-v640-版本开始引入) 和 [`auth-token-refresh-interval`](/tidb-configuration-file.md#auth-token-refresh-interval-从-v640-版本开始引入)。
+2. 定期更新保存 JWKS 至 `auth-token-jwks` 指定的路径。
+3. 创建使用 `tidb_auth_token` 认证的用户，并根据需要通过 `REQUIRE TOKEN_ISSUER` 和 `ATTRIBUTE '{"email": "xxxx@pingcap.com"}` 指定 `iss` 与 `email` 信息。
+4. 生成并签发用于认证的 token，通过 mysql 客户端的 `mysql_clear_text` 插件进行认证。
+
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:189]: L192 到 L211 的示例命令是否可以直接放到 L185~ L248 的各个步骤里并说明是“例如”呢，这样避免用户需要同时对照着看上面的步骤和下面的示例

Comment (CbcWestwolf) [reply security-compatibility-with-mysql.md:190]: Done, PTAL

File: security-compatibility-with-mysql.md [hunk 21]
```diff
@@ -130,3 +130,121 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+> - Header 与 Payload 使用 base64 进行编码的过程是可逆的，请勿在 Payload 中携带敏感数据。
+> - `tidb_auth_token` 认证方式要求客户端支持 [`mysql_clear_password`](https://dev.mysql.com/doc/refman/8.0/en/cleartext-pluggable-authentication.html) 插件，并将 token 以明文的方式发送至 TiDB，因此请[为 TiDB 开启加密传输](/enable-tls-between-clients-and-servers.md) 后再使用 `tidb_auth_token` 进行认证。
+
+#### 使用方法
+
+配置并使用 `tidb_auth_token` 作为 TiDB 用户的认证方式，有以下几个步骤：
+
+1. 在 TiDB 配置文件中设置 [`auth-token-jwks`](/tidb-configuration-file.md#auth-token-jwks-从-v640-版本开始引入) 和 [`auth-token-refresh-interval`](/tidb-configuration-file.md#auth-token-refresh-interval-从-v640-版本开始引入)。
+
+    例如，可以通过下列命令获取示例 JWKS:
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:191]:
```suggestion
    例如，可以通过下列命令获取示例 JWKS：
```

File: security-compatibility-with-mysql.md [hunk 22]
```diff
@@ -130,3 +130,121 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+Signature 用于对 Header 和 Payload 这两部分数据进行签名。
+
+> **警告：**
+>
+> - Header 与 Payload 使用 base64 进行编码的过程是可逆的，请勿在 Payload 中携带敏感数据。
+> - `tidb_auth_token` 认证方式要求客户端支持 [`mysql_clear_password`](https://dev.mysql.com/doc/refman/8.0/en/cleartext-pluggable-authentication.html) 插件，并将 token 以明文的方式发送至 TiDB，因此请[为 TiDB 开启加密传输](/enable-tls-between-clients-and-servers.md) 后再使用 `tidb_auth_token` 进行认证。
+
+#### 使用方法
+
+配置并使用 `tidb_auth_token` 作为 TiDB 用户的认证方式，有以下几个步骤：
```
Comment (qiancai) [inline security-compatibility-with-mysql.md:187]:
```suggestion
配置并使用 `tidb_auth_token` 作为 TiDB 自托管环境中用户的认证方式，有以下几个步骤：
```

File: security-compatibility-with-mysql.md [hunk 23]
```diff
@@ -130,3 +130,121 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+    ```
+
+    然后在 TiDB 的配置文件 `config.toml` 中配置上述 JWKS 文件的路径：
+
+    ```toml
+    [security]
+    auth-token-jwks = "JWKS.json"
+    ```
+
+2. 启动 `tidb-server`，并定期更新保存 JWKS 至 `auth-token-jwks` 指定的路径。
```
Comment (hfxsd) [inline security-compatibility-with-mysql.md:204]:
```suggestion
2. 启动 `tidb-server`，并定期更新保存 JWKS 至 `auth-token-jwks` 指定的路径。

```

File: security-compatibility-with-mysql.md [hunk 24]
```diff
@@ -130,3 +130,121 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
...
+    -----BEGIN PUBLIC KEY-----
+    MIIBCgKCAQEAq8G5n9XBidxmBMVJKLOBsmdOHrCqGf17y9+VUXingwDUZxRp2Xbu
+    LZLbJtLgcln1lC0L9BsogrWf7+pDhAzWovO6Ai4Aybu00tJ2u0g4j1aLiDdsy0gy
+    vSb5FBoL08jFIH7t/JzMt4JpF487AjzvITwZZcnsrB9a9sdn2E5B/aZmpDGi2+Is
+    f5osnlw0zvveTwiMo9ba416VIzjntAVEvqMFHK7vyHqXbfqUPAyhjLO+iee99Tg5
+    AlGfjo1s6FjeML4xX7sAMGEy8FVBWNfpRU7ryTWoSn2adzyA/FVmtBvJNQBCMrrA
+    hXDTMJ5FNi8zHhvzyBKHU0kBTS1UNUbP9wIDAQAB
+    -----END PUBLIC KEY-----
+
+    eyJhbGciOiJSUzI1NiIsImtpZCI6InRoZS1rZXktaWQtMCIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InVzZXJAcGluZ2NhcC5jb20iLCJleHAiOjE3MDMzMDU0OTQsImlhdCI6MTcwMzMwNDU5NCwiaXNzIjoiaXNzdWVyLWFiYyIsInN1YiI6InVzZXJAcGluZ2NhcC5jb20ifQ.T4QPh2hTB5on5xCuvtWiZiDTuuKvckggNHtNaovm1F4RvwUv15GyOqj9yMstE-wSoV5eLEcPC2HgE6eN1C6yH_f4CU-A6n3dm9F1w-oLbjts7aYCl8OHycVYnq609fNnb8JLsQAmd1Zn9C0JW899-WSOQtvjLqVSPe9prH-cWaBVDQXzUJKxwywQzk9v-Z1Njt9H3Rn9vvwwJEEPI16VnaNK38I7YG-1LN4fAG9jZ6Zwvz7vb_s4TW7xccFf3dIhWTEwOQ5jDPCeYkwraRXU8NC6DPF_duSrYJc7d7Nu9Z2cr-E4i1Rt_IiRTuIIzzKlcQGg7jd9AGEfGe_SowsA-w
```
Comment (hfxsd) [inline security-compatibility-with-mysql.md:233]:
```suggestion    
 
 
 
 eyJhbGciOiJSUzI1NiIsImtpZCI6InRoZS1rZXktaWQtMCIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InVzZXJAcGluZ2NhcC5jb20iLCJleHAiOjE3MDMzMDU0OTQsImlhdCI6MTcwMzMwNDU5NCwiaXNzIjoiaXNzdWVyLWFiYyIsInN1YiI6InVzZXJAcGluZ2NhcC5jb20ifQ.T4QPh2hTB5on5xCuvtWiZiDTuuKvckggNHtNaovm1F4RvwUv15GyOqj9yMstE-wSoV5eLEcPC2HgE6eN1C6yH_f4CU-A6n3dm9F1w-oLbjts7aYCl8OHycVYnq609fNnb8JLsQAmd1Zn9C0JW899-WSOQtvjLqVSPe9prH-cWaBVDQXzUJKxwywQzk9v-Z1Njt9H3Rn9vvwwJEEPI16VnaNK38I7YG-1LN4fAG9jZ6Zwvz7vb_s4TW7xccFf3dIhWTEwOQ5jDPCeYkwraRXU8NC6DPF_duSrYJc7d7Nu9Z2cr-E4i1Rt_IiRTuIIzzKlcQGg7jd9AGEfGe_SowsA-w
```

File: security-compatibility-with-mysql.md [hunk 25]
```diff
@@ -108,9 +108,9 @@ TiDB 的密码重用策略功能与 MySQL 一致，在实现密码重用策略
 
 ## 可用的身份验证插件
 
-TiDB 支持多种身份验证方式。通过使用 [`CREATE USER`](/sql-statements/sql-statement-create-user.md) 语句和 [`ALTER USER`](/sql-statements/sql-statement-create-user.md) 语句，即可创建新用户或更改 TiDB 权限系统内的已有用户。TiDB 身份验证方式与 MySQL 兼容，其名称与 MySQL 保持一致。
+TiDB 支持多种身份验证方式。通过使用 [`CREATE USER`](/sql-statements/sql-statement-create-user.md) 语句和 [`ALTER USER`](/sql-statements/sql-statement-alter-user.md) 语句，即可创建新用户或更改 TiDB 权限系统内的已有用户。TiDB 身份验证方式与 MySQL 兼容，其名称与 MySQL 保持一致。
 
-TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务器和客户端建立连接时，如要指定服务器对外通告的默认验证方式，可通过 [`default_authentication_plugin`](/system-variables.md#default_authentication_plugin) 变量进行设置。`tidb_sm3_password` 为仅在 TiDB 支持的 SM3 身份验证方式，使用该方式登录的用户需要使用 [TiDB-JDBC](https://github.com/pingcap/mysql-connector-j/tree/release/8.0-sm3)。`tidb_auth_token` 为仅用于 TiDB Cloud 内部的基于 JSON Web Token (JWT) 的认证方式。
+TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务器和客户端建立连接时，如要指定服务器对外通告的默认验证方式，可通过 [`default_authentication_plugin`](/system-variables.md#default_authentication_plugin) 变量进行设置。`tidb_sm3_password` 为仅在 TiDB 支持的 SM3 身份验证方式，使用该方式登录的用户需要使用 [TiDB-JDBC](https://github.com/pingcap/mysql-connector-j/tree/release/8.0-sm3)。`tidb_auth_token` 为用于 TiDB Cloud 内部的基于 JSON Web Token (JWT) 的认证方式，用户通过配置也可以在自托管环境使用。
```
Comment (hfxsd) [inline security-compatibility-with-mysql.md:113]:
```suggestion
TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务器和客户端建立连接时，如要指定服务器对外通告的默认验证方式，可通过 [`default_authentication_plugin`](/system-variables.md#default_authentication_plugin) 变量进行设置。`tidb_sm3_password` 为仅在 TiDB 支持的 SM3 身份验证方式，使用该方式登录的用户需要使用 [TiDB-JDBC](https://github.com/pingcap/mysql-connector-j/tree/release/8.0-sm3)。`tidb_auth_token` 用于 TiDB Cloud 内部的基于 JSON Web Token (JWT) 的认证，用户通过配置也可以用于自托管环境。
```

**Review by hfxsd — APPROVED** [review]

**Review by qiancai — APPROVED** [review]

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**